frontend/src/metabase/querying/notebook/components/NotebookDataPicker/EmbeddingDataPicker/EmbeddingDataPicker.tsx

```diff
@@ -74,14 +89,22 @@ export function EmbeddingDataPicker({
     );
   }
 
+  const isSourceSelected = Boolean(pickerInfo?.tableId);
   return (
     <PLUGIN_EMBEDDING.DataSourceSelector
-      key={pickerInfo?.tableId}
-      isInitiallyOpen={!table}
+      key={
+        isSourceSelected
+          ? pickerInfo?.tableId
+          : `${sourceTable?.id}:${isSourceModelFetching}`
+      }
+      isInitiallyOpen={isSourceModelFetching ? false : !table}
```

Comment:
we don't wanna open the popover yet until the data source card is loaded since that's when we will know the source model collection ID.

This only happens when you change the starting model without already select data in the join step.